operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -168,7 +168,13 @@ func (r *gatewayReconciler) ensureService(ctx context.Context, desired *corev1.S
 
 	temp := existing.DeepCopy()
 	temp.Spec = desired.Spec
-	temp.SetAnnotations(desired.GetAnnotations())
+	if temp.GetAnnotations() == nil {
+		temp.SetAnnotations(desired.GetAnnotations())
+	} else {
+		for key, value := range desired.GetAnnotations() {
+			temp.Annotations[key] = value
+		}
+	}
```

Comment:
while you are here, can you help to do the same for labels, also for other resources such as CEC and Endpoints? I think it's better to have one common utility function such as mergeMap().

Reply:
I updated this for all types in this file.